x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.ts

```diff
@@ -261,7 +263,48 @@ export const CredentialsLogic = kea<
         flashAPIErrors(e);
       }
     },
-    // TODO onApiTokenChange from ent-search
+    onApiTokenChange: async () => {
+      const { myRole } = AppLogic.values;
+      const {
+        id,
+        name,
+        engines,
+        type,
+        read,
+        write,
+        access_all_engines: accessAllEngines,
+      } = values.activeApiToken;
+
+      const data: IApiToken = {
+        name,
+        type,
+      };
+      if (type === ApiTokenTypes.Private) {
+        data.read = read;
+        data.write = write;
+      }
+      if (type !== ApiTokenTypes.Admin) {
+        data.access_all_engines = !!(accessAllEngines && myRole.canAccessAllEngines);
```

Comment:
I think we need to include a negative test for `access_all_engines`.

Reading the current spec, I would think "Oh, this logic just sets data. access_all_engines to whatever activeApiToken.access_all_engines is", when in fact, it's actually doing more than that, there's an additional change for myRole.canAccessAllEngines which isn't really covered.

Reply:
That's a good point - although now that you point it out, aren't we already checking (and testing) for this value via [fullEngineAccessChecked](https://github.com/elastic/kibana/pull/81519/files#diff-b3a737c6f13a78389c937b1f8a0f7febd716c896dbe47ef611d77ede6d76e19eR1277)?... shouldn't I just reuse that value here (and let the already existing test for `fullEngineAccessChecked` do the work for me) instead of rewriting those tests twice?

Reply:
That sounds reasonable.

Reply:
It's also just weird to me ... like, how could you even get yourself into a situation where you have `access_all_engines` checked in your token, but you DON'T have the corresponding role. I guess perhaps if you had a token previously with that set and your role was changed after then token was created 🤷 

Either way, I agree, I think we could just call out to that selector.

Reply:
Addressed in https://github.com/elastic/kibana/pull/81519/commits/2b200a0520f44c40967e139ea45eb4f4433418f8 - while writing this test I also caught that `fullEngineAccessChecked` is occasionally also sometimes undefined(!) instead of false when myRole is undefined/an empty obj. I fixed the `!!` coercion accordingly and updated expected test values. Woohoo, unit testing :)

Reply:
Sweeeet, this is great ty.